src/promptflow/promptflow/_utils/execution_utils.py

```diff
@@ -56,3 +57,15 @@ def get_aggregation_inputs_properties(flow: Flow) -> AbstractSet[str]:
 def collect_lines(indexes: List[int], kvs: Mapping[str, List]) -> Mapping[str, List]:
     """Collect the values from the kvs according to the indexes."""
     return {k: [v[i] for i in indexes] for k, v in kvs.items()}
+
+
+def extract_aggregation_inputs(flow: Flow, nodes_outputs: dict) -> Dict[str, Any]:
+    """Extract the aggregation inputs of a flow from the nodes outputs."""
+    _aggregation_inputs_references = get_aggregation_inputs_properties(flow)
+    return {prop: _extract_aggregation_input(nodes_outputs, prop) for prop in _aggregation_inputs_references}
+
+
+def _extract_aggregation_input(nodes_outputs: dict, aggregation_input_property: str):
```

Comment:
Optional: change the name for this function.

Reply:
Changed to '_parse_aggregation_input'.